Fixes ChatScreen calling .setScreen

RealMangorage opened the pull request:
Fixes ChatScreen calling .setScreen when we already did .setScreen via ClientCommand for example.

Old Behavior:
/cmd openScreen -> 
Opens screen
ChatScreen.java then proceeds to .setScreen to null


New behavior:
/cmd openScreen -> 
Opens screen
ChatScreen.java will only .setScreen to null if minecraft.screen == this (ChatScreen)

RealMangorage committed:
Fixes ChatScreen calling .setScreen

Fixes ChatScreen calling .setScreen when we already did .setScreen via ClientCommand for example.

Old Behavior:
/cmd openScreen -> 
Opens screen
ChatScreen.java then proceeds to set Null for screen

New behavior:
/cmd openScreen -> 
Opens screen
ChatScreen.java will only .setScreen to null if minecraft.screen == this (ChatScreen)

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/MinecraftForge/MinecraftForge?pullRequest=9443) <br/>All committers have signed the CLA.

sciwhiz12 reviewed (changes requested):
Tested to work with a small modification to `ClientCommandTest` that adds a subcommand that runs `Minecraft.getInstance().setScreen(new PauseScreen(false))`.

However, it seems that `InBedChatScreen` has the same unpatched call in its `keyPressed` method. That should also be patched with this fix, unless someone can provide a reason to leave it unpatched.

RealMangorage committed:
Updated ChatScreen.java

Moves where we do the minecraft.screen == this check.

RealMangorage commented:
I moved the check into .handleChatInput() for minimal code changes.

sciwhiz12 reviewed (changes requested):
Based on the patch hunk header and the CI patch application failure, that patch ain't right.

Please regenerate the patch, and commit it properly rather than using the GitHub web editor.

RealMangorage commented:
Okay

RealMangorage committed:
Test Commit

RealMangorage committed:
Fixes Patch errors

sciwhiz12 commented on patches/minecraft/net/minecraft/client/gui/screens/ChatScreen.java.patch:
```diff
@@ -0,0 +1,11 @@
+          }
+ 
+-         return true;
++         return f_96541_.f_91080_ == this; // To Prevent closing the Screen if it isn't Chat Screen
```
This marker comment should be prefixed with `FORGE:`, and the capitalization be normalized to sentence case.
```suggestion
+         return f_96541_.f_91080_ == this; // FORGE: Prevent closing the screen if another screen has been opened
```

sciwhiz12 commented on src/main/java/net/minecraftforge/server/LanguageHook.java:
```diff
@@ -8,6 +8,7 @@
 import com.google.gson.Gson;
 import com.google.gson.JsonElement;
 import com.google.gson.JsonObject;
+import com.sun.jna.platform.unix.X11;
```
Please revert this change.

sciwhiz12 commented on src/test/java/net/minecraftforge/PatchScreen.java:
```diff
@@ -0,0 +1,17 @@
+package net.minecraftforge;
```
Missing license header, please run the `updateLicenses`/`licenseFormat` task.

Additionally, move this to under the `.debug.client` subpackage please.

sciwhiz12 commented on src/test/java/net/minecraftforge/PatchScreen.java:
```diff
@@ -0,0 +1,17 @@
+import net.minecraft.client.gui.screens.Screen;
+import net.minecraft.network.chat.Component;
+
+public class PatchScreen extends Screen {
```
Formatting: braces on newlines, all throughout this class.

sciwhiz12 commented on src/test/java/net/minecraftforge/PatchScreen.java:
```diff
@@ -0,0 +1,17 @@
+import net.minecraft.network.chat.Component;
+
+public class PatchScreen extends Screen {
+    public PatchScreen(Component p_96550_) {
```
Please rename all the SRG parameters in this class to proper parameter names.

sciwhiz12 commented on src/test/java/net/minecraftforge/debug/chat/ClientCommandTest.java:
```diff
@@ -104,6 +106,11 @@ private void init(RegisterClientCommandsEvent event)
+                                }))
+                        .then(Commands.literal("screenpatch")
+                                .executes((stack) -> {
+                                    Minecraft.getInstance().setScreen(new PatchScreen(Component.literal("patch")));
```
To avoid classloading problems with regards to the (implicit) cast of `PatchScreen` to `Screen`, please move this call to `Minecraft#setScreen` to a static method within `PatchScreen`, which this lambda calls.

sciwhiz12 commented on patches/minecraft/net/minecraft/network/CompressionEncoder.java.patch:
```diff
@@ -13,7 +13,7 @@
           friendlybytebuf.writeBytes(p_129453_);
        } else {
-+         if (!DISABLE_PACKET_DEBUG && i > net.minecraft.network.CompressionDecoder.MAXIMUM_UNCOMPRESSED_LENGTH) {
++         if (!DISABLE_PACKET_DEBUG && i > net.minecraft.network.CompressionDecoder.f_182672_) {
```
I've no idea how this got past us. @Matyrobbrt, ideas 👀?

sciwhiz12 commented on src/test/java/net/minecraftforge/debug/world/LoginPacketSplitTest.java:
```diff
@@ -66,6 +66,7 @@
  * To test if the packet is too large simply remove the login packet from the {@link net.minecraftforge.network.filters.ForgeConnectionNetworkFilter}
  * and try connecting again. You should see the connection fail.
  */
+/**
```
@Matyrobbrt why u cause me pain.

Please fix the compile errors and reinstate this test mod by:
- Changing the call to `FriendlyByteBuf#writeWithCodec` into `FriendlyByteBuf#writeJsonWithCodec`; and
- Changing the reference to `com.mojang.bridge.game.PackType.DATA` to simply `PackType.SERVER_DATA`, importing the new class.

RealMangorage committed:
Renamed PatchScreen.java to TestScreen.java
Fixed LoginPacketSplitTest.java
Update TestScreen.java command in ClientCommandTest.java

RealMangorage commented:
I pushed a commit that fixes that.

sciwhiz12 commented on src/test/java/net/minecraftforge/debug/client/TestScreen.java:
```diff
@@ -0,0 +1,27 @@
+import com.mojang.blaze3d.vertex.PoseStack;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.gui.screens.Screen;
+import net.minecraft.network.chat.Component;
+public class TestScreen extends Screen
```
Formatting: space between last import statement and class declaration.
```suggestion
import net.minecraft.network.chat.Component;

public class TestScreen extends Screen
```

sciwhiz12 commented on src/test/java/net/minecraftforge/debug/client/TestScreen.java:
```diff
@@ -0,0 +1,27 @@
+import net.minecraft.network.chat.Component;
+public class TestScreen extends Screen
+{
+    public static int open() {
+        Minecraft.getInstance().setScreen(new TestScreen(Component.translatable("screen.patch")));
```
Formatting: braces on newlines. 

Additionally, use a literal component here with readable text, since there is no definition for the `screen.patch` translation key.

```suggestion
    public static int open()
    {
        Minecraft.getInstance().setScreen(new TestScreen(Component.literal("Test screen")));
```

sciwhiz12 commented on src/test/java/net/minecraftforge/debug/client/TestScreen.java:
```diff
@@ -0,0 +1,27 @@
+{
+    public static int open() {
+        Minecraft.getInstance().setScreen(new TestScreen(Component.translatable("screen.patch")));
+        return 1;
```
Use `Commands.SINGLE_SUCCESS` here.

sciwhiz12 commented on src/test/java/net/minecraftforge/debug/client/TestScreen.java:
```diff
@@ -0,0 +1,27 @@
+        return 1;
+    }
+
+    public TestScreen(Component pComponent) {
```
Formatting: braces on newlines. Additionally, there shouldn't be a prefix for the parameters.
```suggestion
    public TestScreen(Component title)
    {
```

sciwhiz12 commented on src/test/java/net/minecraftforge/debug/client/TestScreen.java:
```diff
@@ -0,0 +1,27 @@
+    }
+
+    @Override
+    public void render(PoseStack pPoseStack, int pMouseX, int pMouseY, float pPartialTick) {
```
Formatting: braces on newlines. Additionally, there shouldn't be a prefix for the parameters.
```suggestion
    public void render(PoseStack poseStack, int mouseX, int mouseY, float partialTick)
    {
```

RealMangorage committed:
Fixed Bracing for TestScreen.java
